src/Motor.Extensions.Hosting.RabbitMQ/RabbitMQQueueMonitor.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Motor.Extensions.Diagnostics.Queue.Abstractions;
+using Motor.Extensions.Hosting.RabbitMQ.Options;
+using static Motor.Extensions.Hosting.RabbitMQ.LogEvents;
+
+namespace Motor.Extensions.Hosting.RabbitMQ
+{
+    public class RabbitMQQueueMonitor<T> : RabbitMQConnectionHandler, IQueueMonitor where T : notnull
+    {
+        private readonly IOptions<RabbitMQConsumerOptions<T>> _options;
+
+        public RabbitMQQueueMonitor(
+            ILogger<RabbitMQQueueMonitor<T>> logger,
+            IOptions<RabbitMQConsumerOptions<T>> config,
+            IRabbitMQConnectionFactory connectionFactory
+        )
+        {
+            _options = config;
+            _logger = logger;
+            ConnectionFactory = connectionFactory.From(config.Value);
+            EstablishConnection();
+            EstablishChannel();
```

Comment:
fair enough. I'll look into it. I already discussed with @farodin91 how to probably re-use the connection to avoid multiple connections for one consumer but we couldn't think of a good solution yet

Reply:
I reworked the whole RabbitMQ connection factory and friends to be able to share established connections if wanted.
Now the Monitor does not occur as another consumer.
Furthermore I'm exporting the number of current consumers now, too.